### ISSUE TITLE
System for explicitly disabling UI

### ISSUE THREAD

**Issue by mtschoen-unity** [Unity-Technologies/EditorVR#298]
As @dunity and I were discussing #297, we decided that an explicit interface/system for disabling UI interaction was necessary, rather than basing the state on ray/cone visibility. The proposed system:

```
public interface IBlockUI
{
    void SetUIBlockedForRayOrigin(Transform rayOrigin, bool blocked, int priority)
}
```
in EditorVR.Rays.cs:

```
Dictionary<Transform, bool> m_BlockedUIRays;
```
... in AddRaycastSource->isValid:
```
if (m_BlockedUIRays.TryGetValue(rayOrigin, out blocked) && blocked)
    return false;
```

**Comment by dunity** [comment]
Addressed in #297

**Comment by dunity** [comment]
Improvement added, PR merged.
https://github.com/Unity-Technologies/EditorVR/pull/297
